### PR TITLE
`Integrated Code Lifecycle`: imporve auxiliary repository tool tip

### PR DESCRIPTION
**Pull request by SimonEntholzer**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

#### Changes affecting Programming Exercises
- [ ] **High priority**: I tested **all** changes and their related features with **all** corresponding user types on a test server configured with the **integrated lifecycle setup** (LocalVC and LocalCI).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server configured with **Gitlab** and **Jenkins**.

### Motivation and Context

We want the same behaviour for all repository types, on how they are injected before the build process.

### Description

Add a tooltip informing how auxiliary repositories are injected.

### Steps for Testing

Create a new porgramming exercise and check the tool tip for auxiliary repositories

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review

- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance even for very large courses with more than 2000 students.
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance even for very large courses with more than 2000 students.
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Screenshots

image
image

## Summary by CodeRabbit

- **New Features**
	- Enhanced tooltips for programming exercise information with a maximum width for better readability.
	- Updated descriptions for auxiliary repositories in both German and English localization, providing clearer instructions on their usage and behavior during the build process.

- **Bug Fixes**
	- Corrected grammatical inaccuracies in the German description of auxiliary repositories.

### CODE REVIEW

**Review by sachmii — APPROVED** [review]
Tested on TS2, tooltip shows up and works fine.

File: src/main/webapp/i18n/en/programmingExercise.json [hunk 1]
```diff
@@ -519,7 +519,7 @@
             "auxiliaryRepository": {
                 "error": "There is a problem with the auxiliary repository.",
                 "addAuxiliaryRepository": "Add Auxiliary Repository",
-                "usageDescription": "You can use auxiliary repositories to provide additional code that students cannot see or modify. The additional code is inserted at the specified Checkout Directory during the build of the submission.",
+                "usageDescription": "You can use auxiliary repositories to provide additional code that students cannot see or modify. The additional code is inserted at the specified Checkout Directory before the build of the submission. It is inserted via a mv operation. This means, there is a difference on how it is inserted, depending whether the checkout dir already exists or not. If it exists, the repository's content is added at the checkout dir in a new folder with the repository's name. If it does not exist, the checkout dir's folder is created and the auxiliary repositories content is inserted directly there. To overwrite student's files, you need to adapt the build script.",
```
Comment (edkaya) [inline src/main/webapp/i18n/en/programmingExercise.json:522]:
```suggestion
                "usageDescription": "You can use auxiliary repositories to provide additional code that students cannot see or modify. The additional code is inserted into the specified checkout directory before the submission is built, using an mv operation. This means, there is a difference on how it is inserted, depending on whether the checkout directory already exists or not. If it exists, the repository's content is added as a new folder within the checkout directory with the repository's name. If it does not exist, the checkout directory's folder is created and the auxiliary repositories content is inserted directly into it. To overwrite student's files, you need to adapt the build script.",
```

**Review by edkaya — CHANGES_REQUESTED** [review]
LGTM. I left some small adjustments to improve the clarity of the text

**Review by HawKhiem — APPROVED** [review]
Tested on TS3. Works as expected!

image

**Review by SindiBuklaji — DISMISSED** [review]
Tested on TS4. Works as expected & described! 👍

**Review by ole-ve — DISMISSED** [review]
code and tested on TS4

**Review by flbrgit — DISMISSED** [review]
Tested on TS2 and everything works as described and expected

File: src/main/webapp/i18n/en/programmingExercise.json (hunk 1 above)
Comment (SimonEntholzer) [reply src/main/webapp/i18n/en/programmingExercise.json:538]: I added this suggestion :+1:

**Comment by SimonEntholzer** [comment]
https://github.com/ls1intum/Artemis/pull/9585 implements the tooltip update

**Pull request closed without merging**